src/Features/CSharp/Portable/ConvertAutoPropertyToFullProperty/CSharpConvertAutoPropertyToFullPropertyCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,226 @@
+            var accessorListSyntax = ((PropertyDeclarationSyntax)property).AccessorList;
+            var existingAccessors = GetExistingAccessors(accessorListSyntax);
+
+            var getAccessorStatement = generator.ReturnStatement(generator.IdentifierName(fieldName));
+            var newGetter = GetUpdatedAccessor(
+                options, existingAccessors.getAccessor,
+                getAccessorStatement, generator);
+
+            SyntaxNode newSetter = null;
+            if (existingAccessors.setAccessor != null)
+            {
+                var setAccessorStatement = generator.ExpressionStatement(generator.AssignmentStatement(
+                    generator.IdentifierName(fieldName),
+                    generator.IdentifierName("value")));
+                newSetter = GetUpdatedAccessor(
+                    options, existingAccessors.setAccessor,
+                    setAccessorStatement, generator);
+            }
+
+            return (newGetAccessor: newGetter, newSetAccessor: newSetter);
+        }
+
+        private (AccessorDeclarationSyntax getAccessor, AccessorDeclarationSyntax setAccessor) 
+            GetExistingAccessors(AccessorListSyntax accessorListSyntax)
+            => (accessorListSyntax.Accessors.FirstOrDefault(a => a.IsKind(SyntaxKind.GetAccessorDeclaration)),
+                accessorListSyntax.Accessors.FirstOrDefault(a => a.IsKind(SyntaxKind.SetAccessorDeclaration)));
+
+        private SyntaxNode GetUpdatedAccessor(DocumentOptionSet options, 
+            SyntaxNode accessor, SyntaxNode statement, SyntaxGenerator generator)
+        {
+            var newAccessor = AddStatement(accessor, statement);
+            var accessorDeclarationSyntax = (AccessorDeclarationSyntax)newAccessor;
+
+            var preference = GetAccessorExpressionBodyPreference(options);
+            if (preference == ExpressionBodyPreference.Never)
+            {
+                return accessorDeclarationSyntax.WithSemicolonToken(default);
+            }
+
+            if (!accessorDeclarationSyntax.Body.TryConvertToExpressionBody(
+                accessorDeclarationSyntax.Kind(), accessor.SyntaxTree.Options, preference,
+                out var arrowExpression, out var semicolonToken))
+            {
+                return accessorDeclarationSyntax.WithSemicolonToken(default);
+            };
+
+            return accessorDeclarationSyntax
+                .WithExpressionBody(arrowExpression)
+                .WithBody(null)
+                .WithSemicolonToken(accessorDeclarationSyntax.SemicolonToken)
+                .WithAdditionalAnnotations(Formatter.Annotation);
+        }
+
+        internal SyntaxNode AddStatement(SyntaxNode accessor, SyntaxNode statement)
+        {
+            var blockSyntax = SyntaxFactory.Block(
+                SyntaxFactory.Token(SyntaxKind.OpenBraceToken).WithLeadingTrivia(SyntaxFactory.CarriageReturnLineFeed),
```

Comment:
document that the explicit CRLF is so that the block isn't just kept on a single line by the formatter